File: go/cache/cache.go
package cache

import (
	"github.com/Gerrard-YNWA/code/go/cache/memory"
)

type CacheType int

const (
	MemoryCache = iota
)

type Cache interface {
	Get(key string) interface{}
	Set(key string, val interface{}, ttl int64) error
	Del(key string) error
	Destory()
}

func New(ct CacheType) Cache {
	switch ct {
	case MemoryCache:
		return memory.New()
	default:
		return memory.New()
	}
}

File: cpp/cache/cache.cc
#include "cache.h"

#include "lru_cache.h"

namespace gerrard {
namespace cache {
Cache::Cache(int size) {
}

CachePtr Cache::GetCache(CacheType c, int size) {
    return std::make_shared<LruCache>(size);
}

Cache::~Cache() {
}
}  // namespace cache
}  // namespace gerrard
